libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2278,27 +2317,29 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			auto const& toStructType = dynamic_cast<StructType const &>(_to);
 			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 
-			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
-			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
-			solUnimplementedAssert(fromStructType.location() != DataLocation::Memory, "");
-
-			if (fromStructType.location() == DataLocation::CallData)
-			{
-				body = Whiskers(R"(
-					converted := <abiDecode>(value, calldatasize())
-				)")("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
-					{&toStructType}
-				)).render();
-			}
+			if (fromStructType.location() == toStructType.location() && toStructType.isPointer())
+				body = "converted := value";
 			else
 			{
-				solAssert(fromStructType.location() == DataLocation::Storage, "");
+				solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
+				solUnimplementedAssert(fromStructType.location() != DataLocation::Memory, "");
+
+				if (fromStructType.location() == DataLocation::CallData)
+					body = Whiskers(R"(
```

Comment:
Didn't we say earlier that this only works for ``!dynamically encoded`` structs?

Reply:
(:point_up: [8. Oktober 2020 12:54](https://gitter.im/ethereum/solidity-dev?at=5f7eefca1adcf94d3f0ae5e2) and following I mean)

Reply:
Added back assertion

Reply:
But in the wrong place - it's only needed in the ``CallData`` case I believe :-).

Reply:
Which also brings us back our brackets in the end :-).